src/stdlib/parser.ts

```diff
@@ -0,0 +1,309 @@
+    node = <es.VariableDeclaration> node
+    if (node.kind === 'let') {
+      return ({
+        tag: "variable_declaration",
+        name: transform(node.declarations[0].id),
+        value: transform(node.declarations[0].init as es.Expression)
+      })
+    } else if (node.kind === 'const') {
+      return ({
+        tag: "constant_declaration",
+        name: transform(node.declarations[0].id),
+        value: transform(node.declarations[0].init as es.Expression)
+      })
+    } else {
+      unreachable()
+      throw new ParseError("Invalid declaration kind")
+    }
+  }],
+  ["ReturnStatement", (node: es.Node) => {
+    node = <es.ReturnStatement> node
+    return ({
+      tag: "return_statement",
+      expression: transform(node.argument as es.Expression)
+  })}],
+  ["CallExpression", (node: es.Node) => {
+    node = <es.CallExpression> node
+    return ({
+      tag: "application",
+      operator: transform(node.callee),
+      operands: vector_to_list(node.arguments.map(transform))
+  })}],
+  ["UnaryExpression", (node: es.Node) => {
+    node = <es.UnaryExpression> node
+    let loc = <es.SourceLocation> node.loc
+    return ({
+      tag: "application",
+      operator: {
+        tag: "name",
+        name: node.operator,
+        loc: <es.SourceLocation> {
+          start: loc.start,
+          end: { line: loc.start.line, column: loc.start.column + 1 }
+        }
+      },
+      operands: vector_to_list([
+        transform(node.argument)
+      ])
+  })}],
+  ["BinaryExpression", (node: es.Node) => {
+    node = <es.BinaryExpression> node
+    let loc = <es.SourceLocation> node.right.loc
+    return ({
+      tag: "application",
+      operator: {
+        tag: "name",
+        name: node.operator,
+        loc: <es.SourceLocation> {
+          start: { line: loc.start.line, column: loc.start.column - 1 },
+          end: { line: loc.start.line, column: loc.start.column }
+        }
+      },
+      operands: vector_to_list([
+        transform(node.left),
+        transform(node.right)
+      ])
+  })}],
+  ["LogicalExpression", (node: es.Node) => {
+    node = <es.LogicalExpression> node
+    let loc = <es.SourceLocation> node.right.loc
+    return ({
+      tag: "boolean_operation",
+      operator: {
+        tag: "name",
+        name: node.operator,
+        loc: <es.SourceLocation> {
+          start: { line: loc.start.line, column: loc.start.column - 1 },
+          end: { line: loc.start.line, column: loc.start.column }
+        }
+      },
+      operands: vector_to_list([
+        transform(node.left),
+        transform(node.right)
+      ])
+  })}],
+  ["ConditionalExpression", (node: es.Node) => {
+    node = <es.ConditionalExpression> node
+    return ({
+      tag: "conditional_expression",
+      predicate: transform(node.test),
+      consequent: transform(node.consequent),
+      alternative: transform(node.alternate)
+  })}],
+  ["FunctionExpression", (node: es.Node) => {
```

Comment:
Time to remove FunctionExpression

Reply:
Removed all the transformers you listed as well.